x-pack/plugin/ql/src/test/java/org/elasticsearch/xpack/ql/util/StringUtilsTests.java

```diff
@@ -52,4 +52,7 @@ public void testWildcard() {
         assertEquals("^foo\\*$", wildcardToJavaPattern("foox*", 'x'));
     }
 
+    public void testEscapedEscape() {
+        assertEquals("^\\\\\\\\$", wildcardToJavaPattern("\\\\\\\\", '\\'));
+    }
```

Comment:
nit: It is hard to make out the exact number of backslashes. Maybe using raw strings (`"""`) would increase readability?